Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Test and build binary
+FROM golang:1.11.4-stretch as intermediate
+RUN go get github.com/golang/mock/gomock
+RUN go install github.com/golang/mock/mockgen
+RUN go get github.com/golang/dep/cmd/dep
+RUN go install github.com/golang/dep/cmd/dep
+
+WORKDIR /go/src/github.com/Nextdoor/pg-bifrost.git/
+
+# Copy over gopkg and get deps. This will ensure that
+# we don't get the deps each time but only when the files
+# change.
+COPY Gopkg.lock Gopkg.toml ./
+RUN dep ensure --vendor-only
+
+COPY . .
+ARG is_ci
+ENV CI=$is_ci
+RUN make test && make build
```

Comment:
Its not clear why you are setting this here - at the minimum, can you document it? Or is it _really_ necessary for the build process to be somehow different on CI than locally?

Reply:
There is a reason, we have a few places that generate code from interfaces. This is ok to do when developing but I want to explicitly disable this in CI so that we only use checked in code. Will add a comment.